pkg/event/factory.go

```diff
@@ -57,26 +59,47 @@ func CreateEventContext(projectConfig config.ProjectConfig) Context {
 	return context
 }
 
-func createImpressionEvent(projectConfig config.ProjectConfig, experiment entities.Experiment,
-	variation entities.Variation, attributes map[string]interface{}) ImpressionEvent {
+func createImpressionEvent(
+	projectConfig config.ProjectConfig,
+	experiment entities.Experiment,
+	variation *entities.Variation,
+	attributes map[string]interface{},
+	flagKey, flagType string,
+) ImpressionEvent {
+
+	metadata := map[string]string{
+		"FLAG_KEY":  flagKey,
+		"FLAG_TYPE": flagType,
+	}
+
+	var variationID string
+	if variation != nil {
+		metadata["VARIAITON_KEY"] = variation.Key
```

Comment:
nit: should be VARIATION_KEY

Reply:
We've exposing the flaw behind open-ended maps :P